packages/driver/cypress/integration/e2e/multi-domain/multi_domain_spec.ts

```diff
@@ -79,145 +9,220 @@ describe('multi-domain', { experimentalSessionSupport: true }, () => {
+          expect(foo).to.equal('foo')
+          expect(bar).to.equal('bar')
+        })
       })
-    })
 
-    it('passes mixed types to callback function', () => {
-      cy.switchToDomain('http://foobar.com:3500', ['foo', 1, true], ([foo, num, bool]) => {
-        expect(foo).to.equal('foo')
-        expect(num).to.equal(1)
-        expect(bool).to.be.true
+      it('passes array to callback function', () => {
+        cy.switchToDomain('http://foobar.com:3500', ['foo', 'bar'], ([foo, bar]) => {
+          expect(foo).to.equal('foo')
+          expect(bar).to.equal('bar')
+        })
       })
-    })
-  })
 
-  describe('errors', () => {
-    // TODO: Proper stack trace printing still needs to be addressed here
-    it('propagates secondary domain errors to the primary that occur within the test', () => {
-      return new Promise((resolve) => {
-        cy.on('fail', (e) => {
-          expect(e.message).to.equal('done is not defined')
-          resolve(undefined)
+      it('passes string to callback function', () => {
+        cy.switchToDomain('http://foobar.com:3500', ['foo'], ([foo]) => {
+          expect(foo).to.equal('foo')
         })
+      })
 
-        cy.switchToDomain('http://foobar.com:3500', () => {
-          // done is not defined on purpose here as we want to test the error gets sent back to the primary domain correctly
-          // @ts-ignore
-          done()
+      it('passes number to callback function', () => {
+        cy.switchToDomain('http://foobar.com:3500', [1], ([num]) => {
+          expect(num).to.equal(1)
         })
       })
-    })
 
-    it('propagates thrown errors in the secondary domain back to the primary w/ done', (done) => {
-      cy.on('fail', (e) => {
-        expect(e.message).to.equal('oops')
-        done()
+      it('passes boolean to callback function', () => {
+        cy.switchToDomain('http://foobar.com:3500', [true], ([bool]) => {
+          expect(bool).to.be.true
+        })
       })
 
-      cy.switchToDomain('http://foobar.com:3500', () => {
-        throw 'oops'
+      it('passes mixed types to callback function', () => {
+        cy.switchToDomain('http://foobar.com:3500', ['foo', 1, true], ([foo, num, bool]) => {
+          expect(foo).to.equal('foo')
+          expect(num).to.equal(1)
+          expect(bool).to.be.true
+        })
       })
     })
 
-    it('propagates thrown errors in the secondary domain back to the primary w/o done', () => {
-      return new Promise((resolve) => {
+    describe('errors', () => {
+    // TODO: Proper stack trace printing still needs to be addressed here
+      it('propagates secondary domain errors to the primary that occur within the test', () => {
+        return new Promise((resolve) => {
+          cy.on('fail', (e) => {
+            expect(e.message).to.equal('done is not defined')
+            resolve(undefined)
+          })
+
+          cy.switchToDomain('foobar.com', () => {
```

Comment:
I believe you'll still need the scheme and port on these tests.

Reply:
```suggestion
          cy.switchToDomain('http://foobar.com:3500', () => {
```